inc/datahandlers/user.php

```diff
@@ -868,6 +868,31 @@ function verify_language()
 	}
 
 	/**
+	 * Verifies if a style is valid for this user or not.
+	 *
+	 * @return boolean True when valid, false when invalid.
+	 */
+	function verify_style()
+	{
+		global $lang;
+
+		$tid = &$this->data['style'];
+
+		if($tid != 0 && !defined('IN_ADMINCP'))
+		{
+			$theme = get_theme($tid);
+
+			if(!is_member($theme['allowedgroups']) && $theme['allowedgroups'] != 'all')
```

Comment:
For better compatibility with plugins/3rd party registrations/etc. I think that `usergroup` and `additionalgroups` should be passed to `is_member()`, just like it's done in the profile fields code: https://github.com/Sama34/mybb/blob/fix-1752/inc/datahandlers/user.php#L51 Otherwise it will be limited to the current user only.